src/services/studio/studio.service.ts

```diff
@@ -125,33 +124,49 @@ class StudioService {
         });
     }
 
-    private deleteFileIfTypeScenario(packageZip: AdmZip, entry: IZipEntry): void {
-        if (entry.entryName.startsWith("nodes/") && entry.entryName.endsWith(".yml")) {
-            const node: NodeExportTransport = parse(entry.getData().toString());
-            if (node.type === "SCENARIO") {
-                packageZip.deleteFile(entry);
+    private deleteScenarioAssets(packageZip: AdmZip): void {
+        packageZip.getEntries().forEach(entry => {
+            if (entry.entryName.startsWith("nodes/") && entry.entryName.endsWith(".yml")) {
+                const node: NodeExportTransport = parse(entry.getData().toString());
+                if (node.type === "SCENARIO") {
+                    packageZip.deleteFile(entry);
+                }
             }
-        }
+        })
     }
 
-    private fixConnectionVariablesIfRootNodeFile(packageZip: AdmZip, entry: IZipEntry, zipName: string, exportedVariables: VariableManifestTransport[]): void {
-        if (entry.name === "package.yml") {
-            const packageKeyAndVersion = zipName.replace(".zip", "").split("_");
-            const connectionVariablesByKey = this.getConnectionVariablesByKeyForPackage(packageKeyAndVersion[0], packageKeyAndVersion[1], exportedVariables);
+    private fixConnectionVariablesForRootNodeFiles(packageZip: AdmZip, zipName: string, exportedVariables: VariableManifestTransport[]): void {
+        packageZip.getEntries().forEach(entry => {
+            if (entry.name === "package.yml") {
+                const packageKeyAndVersion = this.getPackageKeyAndVersion(zipName);
 
-            if (connectionVariablesByKey.size) {
-                const exportedNode: NodeExportTransport = parse(entry.getData().toString());
-                const nodeContent: NodeSerializedContent = parse(exportedNode.serializedContent);
+                const connectionVariablesByKey = this.getConnectionVariablesByKeyForPackage(packageKeyAndVersion.packageKey, packageKeyAndVersion.version, exportedVariables);
 
-                nodeContent.variables = nodeContent.variables.map(variable => ({
-                    ...variable,
-                    metadata: variable.type === PackageManagerVariableType.CONNECTION ?
-                        connectionVariablesByKey.get(variable.key).metadata : variable.metadata
-                }));
+                if (connectionVariablesByKey.size) {
```

Comment:
I think this part can be moved outside of the loop/find, and that way we can avoid finding the file in the first place, if we are not going to make any changes

Reply:
Since this entire service is not very 'pure function oriented' I moved the part where we read the file instead the if instead of extracting to a function. I think this is more readable than passing the same arguments in one more function since we are using the same references due to how AdmZip works. Do you think this is ok?

Reply:
I meant to make the size check before doing the rest of the logic of the function. Which you updated in the subsequent commit. Marking this as resolved.